pkg/cluster/registerWithHive.go

```diff
@@ -0,0 +1,127 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"os"
+	"time"
+
+	hive "github.com/Azure/ARO-RP/pkg/hiveregistrator"
+	"github.com/openshift/installer/pkg/asset/installconfig/azure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (m *manager) collectDataForHive(ctx context.Context) error {
+
+	m.log.Print("collecting registration data from new cluster")
+
+	cv, err := m.configcli.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	clusterid := cv.Spec.ClusterID
+
+	servicePrincipal := azure.Credentials{
+		TenantID:       m.env.TenantID(),
+		SubscriptionID: m.env.SubscriptionID(),
+		ClientID:       m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientID,
+		ClientSecret:   string(m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientSecret),
+	}
+
+	spBytes, err := json.Marshal(servicePrincipal)
+	if err != nil {
+		return err
+	}
+
+	// TODO: What if a customer adds/modifies a pullsecret after the install ?
```

Comment:
Note: for adoption of existing clusters we need to grab it from the cluster as customer can update pull secret.

We also need to firgure out whether we need to keep pull secret in sync with what we have in hive. This is probably out of scope for this PR, but we need to think about it.

Reply:
Agreed. No action at this time.

Reply:
After discussion with the hive team, I will remove the pullsecret completely. The related doc: https://github.com/openshift/hive/blob/master/docs/using-hive.md#example-adoption-clusterdeployment shows a pullsecret in the example cd but not in the text.